### PR TITLE
Add ability receive system info from different sources 

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: resourcemonitor/resourcemonitor.go [hunk 1]
```diff
@@ -602,6 +615,72 @@ func (monitor *ResourceMonitor) processAlerts() {
 	}
 }
 
+func (xen *xenSystemUsage) RunCmd() {
```
Comment (al1img) [inline resourcemonitor/resourcemonitor.go:618]: I don't see where this function is actually called.

File: resourcemonitor/resourcemonitor.go [hunk 2]
```diff
@@ -602,6 +615,72 @@ func (monitor *ResourceMonitor) processAlerts() {
...
+	lines := make(chan xentop.Line, 1)
+	errs := make(chan error, 1)
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
```
Comment (al1img) [inline resourcemonitor/resourcemonitor.go:622]: You don't need to create own context with own cancel function, just pass the common context here.

File: resourcemonitor/resourcemonitor.go (hunk 1 above)
Comment (MykolaSolyanko) [reply resourcemonitor/resourcemonitor.go]: Done. Sorry forgot to add

File: resourcemonitor/resourcemonitor.go (hunk 2 above)
Comment (MykolaSolyanko) [reply resourcemonitor/resourcemonitor.go]: Done. I did not know that context could be used in a different goroutine

**Pull request merged**